queue_job/jobrunner/runner.py

```diff
@@ -343,6 +343,60 @@ def set_job_enqueued(self, uuid):
                 (ENQUEUED, uuid),
             )
 
+    def reset_dead_jobs(self):
+        """
+        Set started jobs but not locked to pending
+
+        A job is locked when it's being executed
+        When a job is killed, it releases the lock
+        """
+        query = """
+            UPDATE
+                queue_job
+            SET
+                state='pending',
+                retry=retry+1
+            WHERE
+                id IN (
+                    SELECT
+                        job.id
+                    FROM
+                        queue_job_locks AS lock
+                        LEFT JOIN
+                            queue_job AS job
+                        ON
+                            lock.id = job.id
+                    WHERE
+                        job.state  = 'started'
+                    FOR UPDATE SKIP LOCKED
+                );
+        """
+        with closing(self.conn.cursor()) as cr:
+            cr.execute(query)
+
+    def clean_job_lock_table(self):
```

Comment:
Is this necessary? Since we have the on delete cascade on that table, it will be cleaned automatically when the jobs are deleted by the vaccum cleaner.